### DISCUSSION TITLE
Access to rejected path parameter

### DISCUSSION THREAD

**Discussion by pnehrer** [tokio-rs/axum#556] [Q&A] [answered]
Hi, 

is there a way to get the original path parameter value (String) that failed to parse and caused a PathParamRejection?

Alternatively, would it be possible to expose the error type from the path parameter parser?

Thanks!

**Comment by davidpdrsn** [comment]
Not quite sure I understand. Can you elaborate?

**Comment by pnehrer** [comment, marked as answer]
Certainly, suppose I have a few path parameters: `/prefix/:param1/infix/:param2` and a handler that accepts them:

```
async fn handle_request(Path((param1, param2)): Path<(SomeDeserializableType, String)>) {
    ...
}
```

To handle parse errors myself, I can wrap the extractor in `Result<Path<...>, PathParamsRejection>` but even if do, PathParamsRejection doesn't really tell me which parameter failed and why -- I mean there's a description of the error, but the error itself doesn't give more details like e.g., serde_json::de::Error would.

E.g., I would like to get the name of the failed parameter (alone, not in a formatted description) and the expected type.

Or maybe I'm not looking in the right place?

**Reply by davidpdrsn** [reply]
Thank you for clarifying.

Unfortunately there is no good way to get that information currently but I definitely think there should be! Do want to file an issue about it?
